dev/tasks/conda-recipes/arrow-cpp/meta.yaml

```diff
@@ -383,33 +377,31 @@ outputs:
         # skip tests that raise SIGINT and crash the test suite
         {% set tests_to_skip = tests_to_skip + " or (test_csv and test_cancellation)" %}  # [linux]
         {% set tests_to_skip = tests_to_skip + " or (test_flight and test_interrupt)" %}  # [linux]
-        # tests that may crash the agent due to out-of-bound memory writes or other risky stuff
-        {% set tests_to_skip = tests_to_skip + " or test_debug_memory_pool" %}            # [aarch64 or ppc64le]
-        # cannot pass -D_LIBCPP_DISABLE_AVAILABILITY to test suite for our older macos sdk
-        {% set tests_to_skip = tests_to_skip + " or test_cpp_extension_in_python" %}      # [osx]
         # skip tests that make invalid(-for-conda) assumptions about the compilers setup
         {% set tests_to_skip = tests_to_skip + " or test_cython_api" %}                   # [unix]
         {% set tests_to_skip = tests_to_skip + " or test_visit_strings" %}                # [unix]
         # skip tests that cannot succeed in emulation
         {% set tests_to_skip = tests_to_skip + " or test_debug_memory_pool_disabled" %}   # [aarch64 or ppc64le]
         {% set tests_to_skip = tests_to_skip + " or test_env_var_io_thread_count" %}      # [aarch64 or ppc64le]
+        # XMinioInvalidObjectName on osx/win: "Object name contains unsupported characters"
+        {% set tests_to_skip = tests_to_skip + " or test_write_to_dataset_with_partitions_s3fs" %}  # [osx or win]
         # vvvvvvv TESTS THAT SHOULDN'T HAVE TO BE SKIPPED vvvvvvv
-        {% set tests_to_skip = tests_to_skip + " or test_extension_to_pandas_storage_type" %}
-        # segfaults on OSX: to investigate ASAP
-        {% set tests_to_skip = tests_to_skip + " or test_flight" %}                       # [osx]
+        # currently broken
+        {% set tests_to_skip = tests_to_skip + " or test_fastparquet_cross_compatibility" %}
```

Comment:
Haven't raised an issue for this yet, but this is consistently failing. Not sure what's the difference between our CI and the one here.

Reply:
Do you know a CI build log that shows the error for this? (the last ones will have this already skipped) 
(to have an idea what is going on here)

Reply:
```
_____________________ test_fastparquet_cross_compatibility _____________________

tempdir = PosixPath('/tmp/pytest-of-conda/pytest-0/test_fastparquet_cross_compati0')

    @pytest.mark.pandas
    @pytest.mark.fastparquet
    @pytest.mark.filterwarnings("ignore:RangeIndex:FutureWarning")
    @pytest.mark.filterwarnings("ignore:tostring:DeprecationWarning:fastparquet")
    def test_fastparquet_cross_compatibility(tempdir):
        fp = pytest.importorskip('fastparquet')
    
        df = pd.DataFrame(
            {
                "a": list("abc"),
                "b": list(range(1, 4)),
                "c": np.arange(4.0, 7.0, dtype="float64"),
                "d": [True, False, True],
                "e": pd.date_range("20130101", periods=3),
                "f": pd.Categorical(["a", "b", "a"]),
                # fastparquet writes list as BYTE_ARRAY JSON, so no roundtrip
                # "g": [[1, 2], None, [1, 2, 3]],
            }
        )
        table = pa.table(df)
    
        # Arrow -> fastparquet
        file_arrow = str(tempdir / "cross_compat_arrow.parquet")
        pq.write_table(table, file_arrow, compression=None)
    
        fp_file = fp.ParquetFile(file_arrow)
        df_fp = fp_file.to_pandas()
>       tm.assert_frame_equal(df, df_fp)

pyarrow/tests/parquet/test_basic.py:741: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
testing.pyx:55: in pandas._libs.testing.assert_almost_equal
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   AssertionError: DataFrame.iloc[:, 3] (column name="d") are different
E   
E   DataFrame.iloc[:, 3] (column name="d") values are different (66.66667 %)
E   [index]: [0, 1, 2]
E   [left]:  [True, False, True]
E   [right]: [False, False, False]
E   At positional index 0, first diff: True != False

testing.pyx:173: AssertionError
```

Reply:
Searching for fastparquet in our code base, I assume that we actually don't have any CI build that includes it ... So we are just not running that test anywhere, whoops.

Now, the cross-compatibility is also tested in the pandas test suite, so maybe it's not too important to have it here as well. Anyway, opened an issue at https://github.com/apache/arrow/issues/37853

Reply:
Although fastparquet reading a column of boolens wrongly for a pyarrow-written file seems a quite serious issue ..

Reply:
Thanks for raising the issue. I agree it does looks potentially serious, but I hadn't looked too closely because it was just one test. The first thing would be to add testing here in CI. I think this PR is fine as is (I try to keep the test skips to an absolute minimum and remove them whenever possible; that then gets picked up by the next recipe sync anyway)

Reply:
I took a closer look at it (could reproduce this locally), and so actually it was good to do so, as this is a problem in fastparquet reading parquet files generated by latest pyarrow, see details at https://github.com/apache/arrow/issues/37853#issuecomment-1733422017

But yes, so skipping for now is perfectly fine.